probe-rs/src/probe/arm_jtag.rs

```diff
@@ -1197,39 +1169,35 @@ impl<Probe: DebugProbe + RawProtocolIo + JTAGAccess + 'static> RawDapAccess for
         address: u8,
         values: &[u32],
     ) -> Result<(), ArmError> {
-        let mut succesful_transfers = 0;
+        let mut successful_transfers = 0;
 
         let mut idle_cycles = std::cmp::max(1, self.swd_settings().num_idle_cycles_between_writes);
 
+        let mut transfers = values
+            .iter()
+            .map(|v| DapTransfer::write(port, address, *v))
+            .collect::<Vec<_>>();
+
         'transfer: for _ in 0..self.swd_settings().num_retries_after_wait {
-            if succesful_transfers == values.len() {
+            let transfers = &mut transfers[successful_transfers..];
+            if transfers.is_empty() {
                 break;
             }
 
-            let mut transfers: Vec<DapTransfer> = values
-                .iter()
-                .skip(succesful_transfers)
-                .map(|v| DapTransfer::write(port, address, *v))
-                .collect();
-
-            let index_offset = succesful_transfers;
+            perform_transfers(self, transfers, idle_cycles)?;
 
-            perform_transfers(self, &mut transfers, idle_cycles)?;
-
-            for (index, result) in transfers.iter().enumerate() {
-                match &result.status {
-                    TransferStatus::Ok => {
-                        succesful_transfers += 1;
-                    }
+            for result in transfers.iter() {
+                match result.status {
+                    TransferStatus::Ok => successful_transfers += 1,
                     TransferStatus::Failed(err) => {
                         tracing::debug!(
                             "Error in access {}/{} of block access: {}",
-                            index_offset + index + 1,
+                            successful_transfers + 1,
```

Comment:
I think we need to add index still. If one of the transfers before fails this will be wrong now, no?

Reply:
There is no separate index now. We increment `successful_transfers` one line up, and that is our index. If one of the transfers fail before this one, we either return anyway and don't get here, or we retry, in which case we have the same number of successful transfers as before. I'm assuming that if we have a failure at some point, the previous transfers were either immediately successful, or retried and became successful, so I think `successful + 1` is an okay algorithm to print the "number after the last successful transfer" :)

I might be missing something, though!

Reply:
Ok, I am ok to merge like this :)